labelbox/data/serialization/labelbox_v1/label.py

```diff
@@ -0,0 +1,236 @@
+
+from pydantic import BaseModel, Field
+
+from ...annotation_types.annotation import (ClassificationAnnotation,
+                                            ObjectAnnotation,
+                                            VideoClassificationAnnotation,
+                                            VideoObjectAnnotation)
+from ...annotation_types.data import RasterData, TextData, VideoData
+from ...annotation_types.label import Label
+from ...annotation_types.ner import TextEntity
+from .classification import LBV1Classifications
+from .objects import LBV1Objects
+
+
+class LBV1LabelAnnotations(LBV1Classifications, LBV1Objects):
+
+    def to_common(
+            self) -> List[Union[ObjectAnnotation, ClassificationAnnotation]]:
+        classifications = LBV1Classifications.to_common(self)
+        objects = LBV1Objects.to_common(self)
+        return [*objects, *classifications]
+
+    @classmethod
+    def from_common(
+        cls, annotations: List[Union[ClassificationAnnotation,
+                                     ObjectAnnotation]]
+    ) -> "LBV1LabelAnnotations":
+
+        objects = LBV1Objects.from_common(
+            [x for x in annotations if isinstance(x, ObjectAnnotation)])
+        classifications = LBV1Classifications.from_common(
+            [x for x in annotations if isinstance(x, ClassificationAnnotation)])
+        return cls(**objects.dict(), **classifications.dict())
+
+
+class LBV1LabelAnnotationsVideo(LBV1LabelAnnotations):
+    frame_number: int = Field(..., alias='frameNumber')
+
+    def to_common(self):
+        classifications = [
+            VideoClassificationAnnotation(value=classification.to_common(),
+                                          frame=self.frame_number,
+                                          name=classification.title)
+            for classification in self.classifications
+        ]
+
+        objects = [
+            VideoObjectAnnotation(value=obj.to_common(),
+                                  keyframe=obj.keyframe,
+                                  classifications=[
+                                      ClassificationAnnotation(
+                                          value=cls.to_common(),
+                                          schema_id=cls.schema_id,
+                                          name=cls.title,
+                                          extra={
+                                              'feature_id': cls.feature_id,
+                                              'title': cls.title,
+                                              'value': cls.value,
+                                          }) for cls in obj.classifications
+                                  ],
+                                  name=obj.title,
+                                  frame=self.frame_number,
+                                  alternative_name=obj.value,
+                                  schema_id=obj.schema_id,
+                                  extra={
```

Comment:
I'm wondering in general how to decide what should be a class attribute and what should be in the extra dict

Reply:
Things that aren't necessary for modeling but we want to store for compatibility reasons. E.g. color, instanceURI, feature_id, value, all are extra information that isn't necessarily going to be used. That being said, this should be updated to make those fields optional. If that data doesn't exist it shouldn't break the serializer.

Reply:
The extra fields are optional when constructing the lbv1 objects and when converting to annotation types these fields will all just be None.